airflow/providers/amazon/aws/operators/emr_containers.py

```diff
@@ -0,0 +1,151 @@
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        name: str,
+        virtual_cluster_id: str,
+        execution_role_arn: str,
+        release_label: str,
+        job_driver: dict,
+        configuration_overrides: Optional[dict] = None,
+        client_request_token: Optional[str] = None,
+        aws_conn_id: str = "aws_default",
+        sleep_time: int = 30,
+        max_tries: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.name = name
+        self.virtual_cluster_id = virtual_cluster_id
+        self.execution_role_arn = execution_role_arn
+        self.release_label = release_label
+        self.job_driver = job_driver
+        self.configuration_overrides = configuration_overrides or {}
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = client_request_token or str(uuid4())
+        self.sleep_time = sleep_time
+        self.max_tries = max_tries
+        self.job_id = None
+
+    @cached_property
+    def hook(self) -> EMRContainerHook:
+        """Create and return an EMRContainerHook."""
+        return EMRContainerHook(
+            self.aws_conn_id,
+            virtual_cluster_id=self.virtual_cluster_id,
+        )
+
+    def execute(self, context: dict) -> Optional[str]:
+        """Run job on EMR Containers"""
+        self.job_id = self.hook.submit_job(
+            self.name,
+            self.execution_role_arn,
+            self.release_label,
+            self.job_driver,
+            self.configuration_overrides,
+            self.client_request_token,
+        )
+        query_status = self.hook.poll_query_status(self.job_id, self.max_tries, self.sleep_time)
+
+        if query_status in EMRContainerHook.FAILURE_STATES:
+            # self.hook.get_state_change_reason(self.query_execution_id)
+            error_message = "BEEP BOOP"
+            raise AirflowException(
+                "EMR Containers job failed. Final state is {}, query_execution_id is {}. Error: {}".format(
+                    query_status, self.job_id, error_message
+                )
+            )
+        elif not query_status or query_status in EMRContainerHook.INTERMEDIATE_STATES:
+            raise AirflowException(
+                "Final state of EMR Containers job is {}. "
+                "Max tries of poll status exceeded, query_execution_id is {}.".format(
+                    query_status, self.job_id
+                )
+            )
+
+        return self.job_id
+
+    def on_kill(self) -> None:
+        """Cancel the submitted job run"""
+        if self.job_id:
+            self.log.info("Stopping job run with jobId - %s", self.job_id)
+            response = self.hook.stop_query(self.job_id)
+            http_status_code = None
+            try:
+                http_status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+            except Exception as ex:  # pylint: disable=broad-except
```

Comment:
Should this be swallowed or rethrown after logging?

Reply:
Good point...my initial thought was that we don't care if there's an exception while killing the job.

What's the "typical" approach if an operator encounters an exception in `on_kill`?

Reply:
```suggestion
            except Exception as ex:
```

Pylint has been removed.